skcosmo/preprocessing/_flexible_scaler.py

```diff
@@ -0,0 +1,592 @@
+    dataset may result in anomalous behavior.
+
+    At the same time, depending on the conditions of the task, it may be necessary
+    to preserve the ratio in the scale between the features (for example, in the
+    case where the feature matrix is something like a covariance matrix), so
+    the standardization should be carried out for the whole matrix,
+    as opposed to the individual columns,
+    as is done in `sklearn.preprocessing.StandardScaler`.
+
+    Parameters
+    ----------
+    with_mean: bool, default=True
+        If True, center the data before scaling. If False, keep the mean intact
+
+    with_std: bool, default=True
+        If True, scale the data to unit variance. If False, keep the variance intact
+
+    column_wise: bool, default=False
+        If True, normalize each column separately. If False, normalize the whole
+        matrix with respect to its total variance.
+
+    rtol: float, default=1e-5
+        The relative tolerance for the optimization: variance is
+        considered zero when it is less than abs(mean) * rtol + atol.
+
+    atol: float, default=0
+        The relative tolerance for the optimization: variance is
+        considered zero when it is less than abs(mean) * rtol + atol.
+
+    copy : bool, default=None
+        Copy the input X or not.
+
+    Attributes
+    ----------
+    n_samples_seen_: int
+        Number of samples in the reference ndarray
+
+    n_features_: int
+        Number of features in the reference ndarray
+
+    mean_ : ndarray of shape (n_features,)
+        The mean value for each feature in the training set.
+        Equal to ndarray of zeros shape (n_features,) when ``with_mean=False``.
+
+    scale_ : ndarray of shape (n_features,), float  or None
+        The scaling factor, ndarray of shape (n_features,)
+        when `column_wise=True` or float  when `column_wise = False`.
+
+    copy : bool, default=None
+        Copy the input X or not.
+
+    Examples
+    --------
+    >>> import numpy
+    >>> from skcosmo.preprocessing import StandardFlexibleScaler
+    >>> X = np.array([[ 1., -2.,  2.],
+    ...      [ -2.,  1.,  3.],
+    ...      [ 4.,  1., -2.]])
+    >>> transformer = StandardFlexibleScaler().fit(X)
+    >>> transformer
+    StandardFlexibleScaler()
+    >>> transformer.transform(X)
+    array([[ 0.        , -0.56195149,  0.28097574],
+           [-0.84292723,  0.28097574,  0.56195149],
+           [ 0.84292723,  0.28097574, -0.84292723]])
+    >>> transformer.scale_ * transformer.transform(X)
+    array([[ 0., -2.,  1.],
+           [-3.,  1.,  2.],
+           [ 3.,  1., -3.]])
+    >>> transformer.scale_ * transformer.transform(X) + transformer.mean_
+        array([[ 1., -2.,  2.],
+               [-2.,  1.,  3.],
+               [ 4.,  1., -2.]])
+    >>>
+    """
+
+    def __init__(
+        self,
+        with_mean=True,
+        with_std=True,
+        column_wise=False,
+        rtol=1e-5,
+        atol=0,
```

Comment:
I'd be inclined to tweak these defaults a little bit. My general intuition is that something like `rtol=0, atol=1.0E-12` might serve better for most cases, but I'm definitely open to discussion.

Reply:
I have no clear preference, @hurricane642?

Reply:
Same, I chose these settings so that I don't have to fix other codes using `flexible_scaler` in parallel.